subsys/bluetooth/shell/gatt.c

```diff
@@ -764,6 +833,98 @@ static int cmd_metrics(const struct shell *shell, size_t argc, char *argv[])
 	return err;
 }
 
+static u8_t get_cb(const struct bt_gatt_attr *attr, void *user_data)
+{
+	struct shell *shell = user_data;
+	u8_t buf[256];
+	ssize_t ret;
+
+	shell_print(shell, "attr %p uuid %s perm 0x%02x", attr,
+		    bt_uuid_str(attr->uuid), attr->perm);
+
+	if (!attr->read) {
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	ret = attr->read(NULL, attr, (void *)buf, sizeof(buf), 0);
+	if (ret < 0) {
+		shell_print(shell, "Failed to read: %d", ret);
+		return BT_GATT_ITER_STOP;
+	}
+
+	shell_hexdump(shell, buf, ret);
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static int cmd_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	u16_t handle;
+
+	handle = strtoul(argv[1], NULL, 16);
+
+	bt_gatt_foreach_attr(handle, handle, get_cb, (void *)shell);
+
+	return 0;
+}
+
+struct set_data {
+	const struct shell *shell;
+	size_t argc;
+	char **argv;
+	int err;
+};
+
+static u8_t set_cb(const struct bt_gatt_attr *attr, void *user_data)
+{
+	struct set_data *data = user_data;
+	u8_t buf[256];
+	int i;
+	ssize_t ret;
+
+	if (!attr->write) {
+		shell_print(data->shell, "Write not supported");
+		data->err = -ENOENT;
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	for (i = 0; i < data->argc; i++) {
+		buf[i] = strtoul(data->argv[i], NULL, 16);
+	}
+
+	ret = attr->write(NULL, attr, (void *)buf, i, 0, 0);
+	if (ret < 0) {
+		data->err = ret;
+		shell_print(data->shell, "Failed to write: %d", ret);
+		return BT_GATT_ITER_STOP;
+	}
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static int cmd_set(const struct shell *shell, size_t argc, char *argv[])
+{
+	u16_t handle;
+	struct set_data data;
+
+	handle = strtoul(argv[1], NULL, 16);
+
+	data.shell = shell;
+	data.argc = argc - 2;
+	data.argv = argv + 2;
+	data.err = 0;
+
+	bt_gatt_foreach_attr(handle, handle, set_cb, &data);
+
+	if (data.err < 0) {
+		return -ENOEXEC;
```

Comment:
you can consider adding `shell_error` with an error information.

Reply:
There are error printed on set_cb.